providers/parameterstore/parameterstore.go

```diff
@@ -0,0 +1,214 @@
+	cb 		func(s string) string
+}
+
+// Provider returns an AWS ParameterStore provider.
+func Provider(cfg Config, cb func(s string) string) *PSConfig {
+	// load the default config
+	c, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return nil
+	}
+
+	// check inputs and set
+	if cfg.Delim == "" {
+		cfg.Delim = "_"
+	}
+	if cfg.AWSRegion != "" {
+		c.Region = cfg.AWSRegion
+	}
+
+	// Check if AWS Access Key ID and Secret Key are specified.
+	if cfg.AWSAccessKeyID != "" && cfg.AWSSecretAccessKey != "" {
+		c.Credentials = credentials.NewStaticCredentialsProvider(cfg.AWSAccessKeyID, cfg.AWSSecretAccessKey, "")
+	}
+
+	// Check if AWS Role ARN is present.
+	if cfg.AWSRoleARN != "" {
+		stsSvc := sts.NewFromConfig(c)
+		credentials := stscreds.NewAssumeRoleProvider(stsSvc, cfg.AWSRoleARN)
+		c.Credentials = aws.NewCredentialsCache(credentials)
+	}
+	client := ssm.NewFromConfig(c)
+
+	return &PSConfig{client: client, config: cfg, cb: cb}
+}
+
+// ProviderWithClient returns an AWS ParameterStore provider
+// using an existing AWS parameterstore client.
+func ProviderWithClient(cfg Config, cb func(s string) string, client *ssm.Client) *PSConfig {
+	return &PSConfig{client: client, config: cfg, cb: cb}
+}
+
+// Read is not supported by the ParameterStore provider.
+func (ps *PSConfig) Read() (map[string]interface{}, error)  {
+
+	// check if parametername is provided
+	if ps.config.Name == "" {
+		return nil, errors.New("no parameter name provided")
+	}
+
+	// set ssm get parameter input
+	ps.input = ssm.GetParameterInput{
+		Name: aws.String(ps.config.Name),
+	}
+
+	// check if latest version exist, then update name value is "Name": "name:version"
+	if ps.config.VersionId != "" {
+		ps.input.Name = aws.String(*ps.input.Name + ":" + ps.config.VersionId)
+	}
+
+	// get parameter value
+	conf, err := ps.client.GetParameter(context.TODO(), &ps.input)
+	if err != nil {
+		return nil, err
+	}
+
+	mp := make(map[string]interface{})
+
+	// check if value is set as string or string slice
+	if (conf.Parameter.Type == types.ParameterTypeString) || (conf.Parameter.Type == types.ParameterTypeStringList) {
+		key := *conf.Parameter.Name
+		// transform key id transformer provided
+		if ps.cb != nil {
+			key = ps.cb(key)
+		}
+		if key == "" {
+			return nil, errors.New("transformed key has become null")
+		}
+		// set key value
+		mp[key] = *conf.Parameter.Value
+	}
+	
+	// if value is set as map it will unfaltten 
+	if ps.config.Type == "map" {
+		// parse secret value as map if type is set as map
+		valueMap, err := json.Parser().Unmarshal([]byte(*conf.Parameter.Value))
+		if err != nil {
+			return nil, errors.New("unable to unmarshal value as obj")
+		}
+		// modify each value
+		for k, v := range valueMap {
+			updated_key := *conf.Parameter.Name + ps.config.Delim + k
+			// transform key id transformer provided
+			if ps.cb != nil {
+				updated_key = ps.cb(updated_key)
+			}
+			// If the callback blanked the key, it should be omitted
```

Comment:
Why is the transformed key omitted here where as on line 141, an empty transformed key throws an error?

Reply:
this is if the parameter value string is provided as map, if any key inside the map becomes null, I thought it would be better to skip it, on like 141 the key(parameter name) itself becomes null. I can throw a error here. what you suggest?

Reply:
This is if the value of parameter is provided as string. While processing map keys if the key becomes null I decided to skip it. On like 141, the parameter name itself becomes null. I could throw an error here too. What you suggest?

Reply:
updated.

Reply:
I'm unable to figure out what has been updated. Here also, it's the key transformation callback that's called, same as line 141. If the transformer returns an empty key, it should throw an error without skipping, right?

Reply:
Yes, It should throw an error, I updated it in following commit. 
https://github.com/knadh/koanf/pull/175/commits/b04d539dc24b320a2ebc0e0f8db6e47875f3722e

Reply:
Yes, It should throw an error. I updated it in following commit.
https://github.com/knadh/koanf/pull/175/commits/b04d539dc24b320a2ebc0e0f8db6e47875f3722e

Reply:
@knadh updates?